### ISSUE TITLE
support for new outlook

### ISSUE THREAD

**Issue by nckbgov** [xeric/alfred-outlook#14]
Great workflow! It's does excellent job for the classical outlook application.
However, if user switches to the new outlook, the search can't find new e-mails received after moving to the new view.

workaround: move back to the old view but new outlook features are lost

![image](https://user-images.githubusercontent.com/5070901/110102116-aa49bd80-7da4-11eb-9254-5427fac138e4.png)

**Comment by xeric** [comment]
After spending a lot of time to find solution that can work on new outlook, I have to fully refactor (or rewrite) whole workflow.
The new Outlook now is not writing message (Mail) information into Sqlite database anymore when you switch it to new version.
I'm starting the design working now, and hope it could get a workable version ASAP.

**Comment by pszgit** [comment]
Let us know if there is an update to great outlook search workflow. Many thanks.

**Comment by xeric** [comment]
@pszgit the quite tough challenge here is there's no documentation for information regard the new Apple Script APIs for new outlook, still in investigation

**Comment by alsbjer** [comment]
Is it possible to donate for this cause?

**Comment by xeric** [comment]
@alsbjer until now, this is just because Microsoft did not update any information how to use new outlook search via applescript (it was storing mail metadata in db and had file based file metadata which we can use mdfind and mdls to get something in old version), now both of them are gone, MS did not provide how new index works, that's the key problem, I have to keep my eyes on community or internet where I can find some information of way to access mail metadata in new outlook, meanwhile I may find some clues with taking some system level analysis.
This project is my personal Interests and hobbies, donate helps nothing, but thanks anyway!

**Comment by giovannicoppola** [comment]
Hi @xeric any luck with this? It seems the new Outlook uses CoreSpotlight when offline and Microsoft search when online. Are there examples of other apps accessing the CoreSpotlight database?

**Comment by snipem** [comment]
@giovannicoppola Is there any option to query CoreSpotlight from external scripts?

**Comment by giovannicoppola** [comment]
no, it sounds like they might implement Applescript at some point, but no concrete plans. Hopefully the old, sqlite-based version will be available for a while. @xeric do you plan to migrate this to Python 3? If not, I am happy to do that and send you a PR.

**Comment by xeric** [comment]
@giovannicoppola  I tried with mdfind and mdls but still could not find latest mail, let me check if I can get something from coresplotlight, regard py3, the only problem is it could not work fine on alfred v3, I did not purchase v4 upgrade, not sure if there's anyway that can make one extension works both on v3 and v4, if you can ensure the backward compatibility, please raise PR:)
